types/stylelint/index.d.ts

```diff
@@ -246,7 +246,7 @@ declare namespace stylelint {
 	/** @internal */
 	export type RuleBase<P = any, S = any> = (
```

Comment:
[suggest] I'm not sure, but is it worth making the `S` default type more specific? For example:

```suggestion
	export type RuleBase<P = any, S = Record<string, any>> = (
```

Reply:
It's slightly more likely to be more error-prone but I agree with you nonetheless.
i.e. it's the de facto convention

Reply:
I don't think it should be the default.
Do you mean `secondaryOptions: S extends Record<string, any> ? S : never,` instead?

Reply:
No. If my suggestion cannot get a value, let's leave it as-is. I was just curious.

Reply:
Are you fine with [17cd6ca](https://github.com/stylelint/stylelint/pull/7950/commits/17cd6ca5eeb35b435539eeb9fd763f9eee600032) or should I revert to `S`.
It's slightly more strict.
I am not sure myself if we want to restrict the type for plugins or not.
i.e. maybe an array could be acceptable